bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/gmos/GmosRecipe.java

```diff
@@ -72,6 +66,18 @@ public GmosRecipe(SourceDefinitionParameters sdParameters,
         _gmosParameters = gmosParameters;
         _teleParameters = teleParameters;
         _plotParameters = plotParameters;
+
+
+        // TODO: this validation should be done centrally somewhere (several instruments do similar stuff here)
+        if (_sdParameters.getDistributionType().equals(SourceDefinitionParameters.Distribution.ELINE)) {
+            if (_sdParameters.getELineWidth() < (3E5 / (_sdParameters.getELineWavelength() * 1000))) {
+                throw new RuntimeException(
+                        "Please use a model line width > 1 nm (or "
+                                + (3E5 / (_sdParameters.getELineWavelength() * 1000))
+                                + " km/s) to avoid undersampling of the line profile when convolved with the transmission response");
+            }
```

Comment:
This is a valid point. So far ITC just has a "catch-all" clause somewhere high up in the call stack that prints the exception message to the html page sent back to the user. I agree there's probably a need for something more sophisticated.

In particular, I've been discussing with Andy the idea to turn errors and suggestions like these into warnings and errors in the OT (using same mechanisms as the P2 checks). If we want to do that, obviously the error handling will have to change.

I haven't made up my mind yet how to attack this, so far I've just been moving around those validations with the rest of the code, so they're still randomly distributed throughout the code base. However, since I am starting now to work on the actual service interface that will be used to access the calculations in the future I'll have to address this sooner than later. I am planning to have the service layer entirely in Scala and only leave the calculation code in Java, that gives us some more possibilities here to use ```Validation```or whatever is most useful here. To be continued...